### PR TITLE
Adding retries to ConsumerLagsResourceIntegrationTest

### PR DESCRIPTION
**Pull request by ahuang98**
Wrapping consumer subscribes and polls in retry block to remove flakiness and address the following test failures

https://confluentinc.atlassian.net/browse/KREST-625?atlOrigin=eyJpIjoiMzVjMzg2Njc5MDQxNDUzYWI2NjVjMWY1YzRhYmJjNmMiLCJwIjoiaiJ9
https://confluentinc.atlassian.net/browse/KREST-616?atlOrigin=eyJpIjoiOGJkYTcxNjZmMjFjNGNkOThhZTM0ZDgxMTBhMTk4ZGIiLCJwIjoiaiJ9

### CODE REVIEW

File: kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/ConsumerLagsResourceIntegrationTest.java [hunk 1]
```diff
@@ -77,67 +78,66 @@ public void setUp() throws Exception {
...
+          consumer2.commitSync();
+
+          // stores expected currentOffsets and logEndOffsets for each topic partition after sending
+          // 3 records to topic1 partition0 and topic2 partition1
+          long[][] expectedOffsets = new long[numTopics][numPartitions];
+          expectedOffsets[0][0] = 3;
+          expectedOffsets[1][1] = 3;
+          // all other values default to 0L
+
+          Response response =
```
Comment (rigelbm) [inline kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/ConsumerLagsResourceIntegrationTest.java:111]: I think you are retrying too much code. Can't you just retry starting from here?

File: kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/ConsumerLagsResourceIntegrationTest.java [hunk 2]
```diff
@@ -210,43 +210,46 @@ public void listConsumerLags_nonExistingConsumerGroup_returnsNotFound() {
...
+          // 3 records to topic1 partition0 and topic2 partition1
+          long[][] expectedOffsets = new long[numTopics][numPartitions];
+          expectedOffsets[0][0] = 3;
+          expectedOffsets[1][1] = 3;
+          // all other values default to 0L
+
+          for (int t = 0; t < numTopics; t++) {
+            for (int p = 0; p < numPartitions; p++) {
+              consumer.seekToEnd(Collections.singletonList(new TopicPartition(topics[t], p)));
+              Response response =
```
Comment (rigelbm) [inline kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/ConsumerLagsResourceIntegrationTest.java:237]: Same here.

File: kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/ConsumerLagsResourceIntegrationTest.java (hunk 1 above)
Comment (ahuang98) [reply kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/ConsumerLagsResourceIntegrationTest.java:111]: I thought the test failures we saw (something along the lines of expected 3 but got 0 for getCurrentOffset() L135) stemmed from the consumer not polling/committing the offsets properly.

Comment (rigelbm) [reply kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/ConsumerLagsResourceIntegrationTest.java:111]: I think there's a delay between whatever the consumer is doing above and the HTTP request below. The retry should only be in the HTTP request, to discard the possibility we just didn't wait long enough for the consumer to do its thing.

**Review by rigelbm — APPROVED** [review]
LGTM

**Pull request merged**